core/src/test/scala/unit/kafka/server/AbstractApiVersionsRequestTest.scala

```diff
@@ -41,12 +42,18 @@ abstract class AbstractApiVersionsRequestTest(cluster: ClusterInstance) {
     } else {
       cluster.brokerSocketServers().asScala.head
     }
-    IntegrationTestUtils.connectAndReceive[ApiVersionsResponse](request, socket, listenerName)
+
+    val port = socket.boundPort(listenerName)
+
+    IntegrationTestUtils.connectAndReceive[ApiVersionsResponse](request, port)
   }
 
   def sendUnsupportedApiVersionRequest(request: ApiVersionsRequest): ApiVersionsResponse = {
     val overrideHeader = IntegrationTestUtils.nextRequestHeader(ApiKeys.API_VERSIONS, Short.MaxValue)
-    val socket = IntegrationTestUtils.connect(cluster.brokerSocketServers().asScala.head, cluster.clientListener())
+    val socketServer = cluster.brokerSocketServers().asScala.head
```

Comment:
Could you please add following helper to `ClusterInstance`?
```
    default Collection<Integer> boundPorts() {
        return brokers().values().stream()
                .map(KafkaBroker::socketServer)
                .map(s -> s.boundPort(clientListener()))
                .collect(Collectors.toList());
    }
```

this helper can streamline this PR I believe